src/main/java/uk/gov/hmcts/ccd/data/user/DefaultUserRepository.java

```diff
@@ -61,6 +67,13 @@ public IDAMProperties getUserDetails() {
         return restTemplate.exchange(applicationParams.idamUserProfileURL(), HttpMethod.GET, requestEntity, IDAMProperties.class).getBody();
     }
 
+    @Override
```

Comment:
Caching this method for 8 hours, should lend benefits on case events. The call to IDAM was being made just to retrieve the user's first name and last name which then is stored in case event. I don't think user details like first name and last name will change often. What do you think?